data/campaigns/tutorial/scenarios/01_Tutorial_part_1.cfg

```diff
@@ -194,6 +210,85 @@
         [/lua]
     [/event]
 
+    # Give a warning to undo last step if target is out of reach
+    # Check whether at least one of our units can reach any of the target_loc locations
+    [event]
+        name=moveto
+        id=check_undo
+        first_time_only=no
+        [filter]
+            side=1
+        [/filter]
+        [allow_undo][/allow_undo]
+
+        # Store all our units
+        [store_unit]
+            [filter]
+                side=1
+            [/filter]
+            variable=student_team
+        [/store_unit]
+
+        {VARIABLE within_reach no}
+        #{VARIABLE mindist 100} # kept for debugging
+
+        [foreach]   # Check each unit
+            array=student_team
+            [do]
+                {VARIABLE moves_left $this_item.moves}
+                {VARIABLE unit_id $this_item.id}
+
+                [foreach]  # compare their movement points with the movement costs of the valid target locations
+                    array=target_loc
+                    [do]
+                        [find_path]
+                            [traveler]
+                                id=$unit_id
+                            [/traveler]
+                            [destination]
+                                x=$this_item.x
+                                y=$this_item.y
+                            [/destination]
+                            variable=curr_distance
+                            allow_multiple_turns=yes
+                            check_zoc=no
```

Comment:
Why allowing multiturn, was it not about the current turn? And maybe zoc can be taken into account ?

Reply:
multiturn: That is because I did not know what will be stored in curr_distance if allow_multiple_turns is disabled and target can't be reached.

zoc: you are right, probably a copy-paste leftover, will fix it.

Reply:
Tried to disable allow_multiple_turns, but then the distance calculation stops working. Had no time to test it further though.